processor/src/test/java/org/mapstruct/ap/test/nestedbeans/DottedErrorMessageTest.java

```diff
@@ -0,0 +1,186 @@
+import org.mapstruct.ap.test.nestedbeans.erroneous.WordDto;
+import org.mapstruct.ap.testutil.WithClasses;
+import org.mapstruct.ap.testutil.compilation.annotation.CompilationResult;
+import org.mapstruct.ap.testutil.compilation.annotation.Diagnostic;
+import org.mapstruct.ap.testutil.compilation.annotation.ExpectedCompilationOutcome;
+import org.mapstruct.ap.testutil.runner.AnnotationProcessorTestRunner;
+
+@WithClasses({
+    Car.class, CarDto.class, Color.class, ColorDto.class,
+    House.class, HouseDto.class, Roof.class, RoofDto.class,
+    User.class, UserDto.class, Wheel.class, WheelDto.class,
+    Dictionary.class, DictionaryDto.class, Word.class, WordDto.class,
+    ForeignWord.class, ForeignWordDto.class,
+    Computer.class, ComputerDto.class,
+    Cat.class, CatDto.class, Info.class, InfoDto.class
+})
+@RunWith(AnnotationProcessorTestRunner.class)
+public class DottedErrorMessageTest {
+
+    private static final String PROPERTY = "property";
+    private static final String COLLECTION_ELEMENT = "Collection element";
+    private static final String MAP_KEY = "Map key";
+    private static final String MAP_VALUE = "Map value";
+
+    @Test
+    @WithClasses({
+        UnmappableDeepNestingMapper.class
+    })
+    @ExpectedCompilationOutcome(
+        value = CompilationResult.FAILED,
+        diagnostics = {
+            @Diagnostic(type = UnmappableDeepNestingMapper.class,
+                kind = javax.tools.Diagnostic.Kind.ERROR,
+                line = 26,
+                messageRegExp = "Can't map " + PROPERTY +
+                    " \".*Color house\\.roof\\.color\" to \".*house\\.roof\\.color\"\\. " +
+                    "Consider to declare/implement a mapping method: \".*ColorDto map\\(.*Color value\\)\"\\.")
+        }
+    )
+    public void testDeepNestedBeans() {
+    }
+
+    @Test
+    @WithClasses({
+        UnmappableDeepListMapper.class
+    })
+    @ExpectedCompilationOutcome(
+        value = CompilationResult.FAILED,
+        diagnostics = {
+            @Diagnostic(type = UnmappableDeepListMapper.class,
+                kind = javax.tools.Diagnostic.Kind.ERROR,
+                line = 26,
+                messageRegExp = "Can't map " + COLLECTION_ELEMENT +
+                    " \".*Wheel car\\.wheels\" to \".*car\\.wheels\"\\. " +
+                    "Consider to declare/implement a mapping method: \".*WheelDto map\\(.*Wheel value\\)\"\\.")
+        }
+    )
+    public void testIterables() {
+    }
+
+    @Test
+    @WithClasses({
+        UnmappableDeepMapKeyMapper.class
+    })
+    @ExpectedCompilationOutcome(
+        value = CompilationResult.FAILED,
+        diagnostics = {
+            @Diagnostic(type = UnmappableDeepMapKeyMapper.class,
+                kind = javax.tools.Diagnostic.Kind.ERROR,
+                line = 26,
+                messageRegExp = "Can't map " + MAP_KEY +
+                    " \".*Word dictionary\\.wordMap\\{:key\\}\" to \".*dictionary\\.wordMap\\{:key\\}\"\\. " +
```

Comment:
I agree with @agudian . Currently we use the same pattern for all messages and it might theoretically confuse the user why there are different error messages for similar cases.